examples/typescript-react/js/app.tsx

```diff
@@ -0,0 +1,175 @@
+/*jshint quotmark:false */
+/*jshint white:false */
+/*jshint trailing:false */
+/*jshint newcap:false */
+/*global React, Router*/
+
+/// <reference path="../typings/tsd.d.ts" />
+declare var Router : any;
+
+var TodoModel = app.models.TodoModel;
+var TodoFooter = app.components.TodoFooter;
+var TodoItem = app.components.TodoItem;
+
+namespace app.components {
+
+  export class TodoApp extends React.Component<IAppProps, IAppState> {
+
+    constructor(props : IAppProps) {
+      super(props);
+      this.state = {
+        nowShowing: app.constants.ALL_TODOS,
+        editing: null
+      };
+    }
+
+    public componentDidMount() {
+      var setState = this.setState;
+      var router = Router({
+        '/': setState.bind(this, {nowShowing: app.constants.ALL_TODOS}),
+        '/active': setState.bind(this, {nowShowing: app.constants.ACTIVE_TODOS}),
+        '/completed': setState.bind(this, {nowShowing: app.constants.COMPLETED_TODOS})
+      });
+      router.init('/');
+    }
+
+    public handleNewTodoKeyDown(event) {
+      if (event.keyCode !== app.constants.ENTER_KEY) {
+        return;
+      }
+
+      event.preventDefault();
+
+      var val = React.findDOMNode<HTMLInputElement>(this.refs["newField"]).value.trim();
+
+      if (val) {
+        this.props.model.addTodo(val);
+        React.findDOMNode<HTMLInputElement>(this.refs["newField"]).value = '';
+      }
+    }
+
+    public toggleAll(event) {
+      var checked = event.target.checked;
+      this.props.model.toggleAll(checked);
+    }
+
+    public toggle(todoToToggle) {
+      this.props.model.toggle(todoToToggle);
+    }
+
+    public destroy(todo) {
+      this.props.model.destroy(todo);
+    }
+
+    public edit(todo) {
+      this.setState({editing: todo.id});
+    }
+
+    public save(todoToSave, text) {
+      this.props.model.save(todoToSave, text);
+      this.setState({editing: null});
+    }
+
+    public cancel() {
+      this.setState({editing: null});
+    }
+
+    public clearCompleted() {
+      this.props.model.clearCompleted();
+    }
+
+    public render() {
+      var footer;
+      var main;
+      var todos = this.props.model.todos;
+
+      var shownTodos = todos.filter(function (todo) {
+        switch (this.state.nowShowing) {
+        case app.constants.ACTIVE_TODOS:
+          return !todo.completed;
+        case app.constants.COMPLETED_TODOS:
+          return todo.completed;
+        default:
+          return true;
+        }
+      }, this);
```

Comment:
Instead of passing `this`, I would use an arrow function. That way the compiler can still type-check against `this`.